pkg/api/server/v1alpha2/logs.go

```diff
@@ -0,0 +1,386 @@
+package server
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/google/cel-go/cel"
+	celenv "github.com/tektoncd/results/pkg/api/server/cel"
+	"github.com/tektoncd/results/pkg/api/server/db/errors"
+	"github.com/tektoncd/results/pkg/api/server/db/pagination"
+	"github.com/tektoncd/results/pkg/api/server/v1alpha2/result"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gorm.io/gorm"
+	"io"
+
+	"github.com/tektoncd/results/pkg/api/server/db"
+	"github.com/tektoncd/results/pkg/api/server/v1alpha2/auth"
+	"github.com/tektoncd/results/pkg/api/server/v1alpha2/log"
+	"github.com/tektoncd/results/pkg/api/server/v1alpha2/record"
+	"github.com/tektoncd/results/pkg/apis/v1alpha2"
+	"github.com/tektoncd/results/pkg/logs"
+	pb "github.com/tektoncd/results/proto/v1alpha2/results_go_proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func (s *Server) GetLog(req *pb.GetLogRequest, srv pb.Logs_GetLogServer) error {
+	parent, res, name, err := record.ParseName(req.GetName())
+	if err != nil {
+		s.logger.Error(err)
+		return status.Error(codes.InvalidArgument, "Invalid Name")
+	}
+
+	if err := s.auth.Check(srv.Context(), parent, auth.ResourceLogs, auth.PermissionGet); err != nil {
+		s.logger.Error(err)
+		return status.Error(codes.Unauthenticated, "Permission denied")
+	}
+
+	rec, err := getRecord(s.db, parent, res, name)
+	if err != nil {
+		s.logger.Error(err)
+		return err
+	}
+	// Check if the input record is referenced in any logs record in the result
+	if rec.Type != v1alpha2.LogRecordType {
+		rec, err = getLogRecord(s.db, parent, res, name)
+		if err != nil {
+			s.logger.Error(err)
+			return err
+		}
+	}
+
+	streamer, logObject, err := log.ToStream(srv.Context(), rec, s.config.LOGS_CHUNK_SIZE, s.config.LOGS_PATH, s.config)
+	if err != nil {
+		s.logger.Error(err)
+		return status.Error(codes.Internal, "Error streaming log")
+	}
+	if logObject.Status.Size == 0 {
+		s.logger.Errorf("no logs exist for %s", req.GetName())
+		return status.Error(codes.NotFound, "Log doesn't exist")
+	}
+
+	writer := logs.NewBufferedWriter(srv, name, s.config.LOGS_CHUNK_SIZE)
+	if _, err = streamer.WriteTo(writer); err != nil {
+		s.logger.Error(err)
+		return status.Error(codes.Internal, "Error streaming log")
+	}
+	_, err = writer.WriteRemain()
```

Comment:
See my suggestion to rename this function `Flush` - debatable if that makes intent clearer. No matter what, however, we should godoc that this function must be called after writes are completed.